modules/consensus/processed_block.go

```diff
@@ -131,23 +138,56 @@ func clampTargetAdjustment(base *big.Rat) *big.Rat {
 	return base
 }
 
+// setChildTarget computes the target of a blockNode's child. All children of a node
+// have the same target.
+func (cs *ConsensusSet) setChildTarget(blockMap *bolt.Bucket, pb *processedBlock) error {
+	// Fetch the parent block.
+	var parent processedBlock
+	parentBytes := blockMap.Get(pb.Block.ParentID[:])
+	err := encoding.Unmarshal(parentBytes, &parent)
+	if build.DEBUG && err != nil {
+		panic(err)
+	}
+
+	if pb.Height%(types.TargetWindow/2) != 0 {
+		pb.ChildTarget = parent.ChildTarget
```

Comment:
target is only changed every 500 blocks. Otherwise the target is the same as the parent target.

When the target does change, you have to reach back through 500 parents to find your parent (can't use the BlockPath because it might be that 500th grandparent isn't actually in the BlockPath - rare case but still an issue)